src/vision/status_quo/alan_builds_a_cache.md

```diff
@@ -0,0 +1,84 @@
+
+If you would like to expand on this story, or adjust the answers to the FAQ, feel free to open a PR making edits (but keep in mind that, as they reflect peoples' experiences, status quo stories [cannot be wrong], only inaccurate). Alternatively, you may wish to [add your own status quo story][htvsq]!
+
+
+## The story
+
+[Alan][] is working on an HTTP server. The server makes calls to some other service. The performance of the downstream service is somewhat poor, so Alan would like to implement some basic caching.
+
+Alan writes up some code which does the caching:
+
+```rust
+async fn get_response(&mut self, key: String) {
+    // Try to get the response from cache
+    if let Some(cached_response) self.cache.get(key) {
+        self.channel.send(cached_response).await;
+        return;
+    }
+
+    // Get the response from the downstream service
+    let response = self.http_client.make_request(key).await;
+    self.channel.send(response).await;
+    
+    // Store the response in the cache
+    self.cache.set(key, response);
+}
+```
+
+Alan is happy with how things are working, but notices every once in a while the downstream service hangs. To prevent that, Alan implements a timeout.
+
+He remembers from the documentation for his favorite runtime that there is the `race` function which can kick off two futures and polls both until one completes. (similar to tokio's [select](https://docs.rs/tokio/1.5.0/tokio/macro.select.html) and async-std's [race](https://docs.rs/async-std/1.9.0/async_std/future/trait.Future.html#method.race) for example).
+
+
+```rust
+runtime::race(timeout(), get_response(key)).await
+```
+
+## The bug
+
+Alan ships to production but after several weeks he notices some users complaining that they receive old data.
+
+Alan looks for help. The compiler unfortunately doesn't provide any hints. He turns to his second best friend clippy, who cannot help either.
+Alan tries debugging. He uses his old friend `println!`. After hours of working through, he notices that sometimes the line that sets the response in the cache never gets called.
+
+## The solution
+
+Alan goes to [Barbara][] and asks why in the world that might be ⁉️
+
+💡 Barbara looks through the code and notices that there is an await point between sending the response over the channel and setting the cache.
+
+Since the `get_response` future can be dropped at each available await point, it may be dropped *after* the http request has been made, but *before* the response has successfully been sent over the channel, thus not executing the remaining instructions in the function.
+
+This means the cache might not be set.
+
+Alan fixes it by setting the cache before sending the result over the channel. 🎉
+
+```rust
+async fn get_response(&mut self, key: String) {
+    // ... cache miss happened here
+
+    // Yield while we wait for the http request to complete
```

Comment:
@rylev @Stupremee do you think using the `Yield` keyword is appropriate here? or is it wrong/misleading?

Reply:
I feel like `yield` implies some sort of explicit loop or waiting process, which is not present here.
I think something simple like "Perform the HTTP request." should be fine

Reply:
given it's the new example, I'm trying to convey that both await points may or may not resume execution, which is why we wanna set the cache there, but yeah `yield` is confusing.

Any idea of a word that would allow us to convey the meaning better ?

Otherwise I'll just remove this example since its just two line swaps away from the first one so i guess it wouldn't be interesting anymore

Reply:
"We perform the HTTP request and our code *might* continue after this `.await` once the HTTP request is complete"

Is probably a bit simpler to understand

Reply:
definitely! thanks!